programs/utilities/showstopper.sc

```diff
@@ -0,0 +1,264 @@
+// By altrisi
+// Configuration is done via the in-game command, and will be saved and restored from a configuration file
+
+
+// Mode definition
+
+// Returns the tick event handler for this mode.
+global_HANDLER = 0;
+// Returns the information provider for this mode.
+// It's a function that takes no arguments and returns a string
+global_INFO_PROVIDER = 1;
+
+// null means no event handler. Given it's nothing, it needs no global_INFO_PROVIDER
+OFF = [
+    null,
+];
+overN = [
+    'overNHandler',
+    _() -> 'Running in average of ' + global_interval + ' over ' + global_argument + 'mspt mode'
+];
+overNSingle = [
+    'overNSingleHandler',
+    _() -> 'Running in single tick over ' + global_argument + 'mspt mode'
+];
+overPercentage = [
+    'overPercentageHandler', 
+    _() -> 'Running in average of ' + global_interval + ' ticks being at least '+(global_argument-1)*100+'% higher than general average mode'
+];
+
+// Mode setter definition
+// Whether this mode requires the extra argument to be passed to the command
+global_REQUIRES_ARGUMENT = 0;
+// Handles setting the argument and returns the mode to be set
+global_ARG_TO_MODE_FUNCTION = 1;
+
+// Holds name -> modeSetter
+global_MODE_MAP = {
+    'off'              -> [false, _(arg, outer(OFF))            -> OFF],
+    'over50mspt'       -> [false, _(arg, outer(overN))          -> (global_argument = 50;  return(overN))],
+    'over50msptsingle' -> [false, _(arg, outer(overNSingle))    -> (global_argument = 50;  return(overNSingle))],
+    'overpercentage'   -> [true,  _(arg, outer(overPercentage)) -> (global_argument = 1 + arg/100; return(overPercentage))],
+    'overN'            -> [true,  _(arg, outer(overN))          -> (global_argument = arg; return(overN))],
+    'overNsingle'      -> [true,  _(arg, outer(overNSingle))    -> (global_argument = arg; return(overNSingle))]
+};
+
+__config() -> {
+    'scope' -> 'global',
+    'commands' -> {
+        '' -> _() 
+            -> print(
+                '----------- [Showstopper Commands] ----------- \n' +
+                'mode <mode> [argument] | Current: ' + global_modeSaveName + '\n' +
+                '  - Sets the listener mode \n' +
+                'freeze [boolean] | Current: ' + global_shouldFreeze + '\n' + 
+                '  - Sets whether the game should be frozen on anomaly \n' +
+                'interval <ticks> | Current: ' + global_interval + '\n' +
+                '  - Sets the ticks to be grouped in non-single modes \n' +
+                'cooldown <ticks> | Current: ' + global_cooldown + '\n' +
+                '  - Sets the cooldown to wait after an anomaly'
+            ),
+        'freeze' -> _()->setFreeze(!global_shouldFreeze),
+        'freeze <bool>' -> 'setFreeze',
+        'interval <ticks>' -> 'setInterval',
+        'cooldown <ticks>' -> 'setCooldown',
+        'mode <mode>' -> ['setMode', ''],
+        'mode <mode> <argument>' -> 'setMode'

```

Comment:
I would tend to be more explicit here. Having the suggester give `10, 25, 50, ..` when that parameter defaults to 50 and refuses the argument is not ideal. It would be better if it didn't suggest them when they are not accepted. 

It also looks like `/showstopper mode over50mspt` is equivalent to `/showstopper mode overN 50`.
And so I am wondering why include the `over50mspt` version. 
You could do something like `/showstopper mode mspt 10` and have `/showstopper mode mspt` default to 50.

I also found the `over50msptSingle vs over50mspt` confusing at first. I would have understood it faster if I saw the other options where average ticks. 

Something like `/showstoper mode <single/average/percentage> <threshold>` might be more clear for the user.